Mathlib/Algebra/Category/ModuleCat/Presheaf/OverColimitsRing.lean

```diff
@@ -0,0 +1,281 @@
+  have eq₁ : r₁ = colimit.ι R j
+      (R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₁)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₂ : r₂ = colimit.ι R j
+      (R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₂)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₃ : r₁ * r₂ = colimit.ι R j
+      (R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₁) *
+       R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₂)) := by
+    rw [map_mul]
+    erw [colimit.w_apply, colimit.w_apply, Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+  rw [eq₃]
+  conv_rhs => rw [eq₁]; rhs; rw [eq₂]
+  rw [rep_smul, rep_smul, rep_smul, smulColimit.mul_smul]
+
+lemma smul_zero (r : colimit (C := RingCat) R) : colimitSMulColimit (M := M) r 0 = 0 := by
+  rw [show r = colimit.ι R (Concrete.indexRepColimit R r) _ by
+    erw [Concrete.ι_repColimit_eq], rep_smul, smulColimit.smul_zero]
+
+lemma smul_add (r : colimit (C := RingCat) R) (m₁ m₂ : colimit (C := AddCommGrp) M.presheaf) :
+    colimitSMulColimit r (m₁ + m₂) = colimitSMulColimit r m₁ + colimitSMulColimit r m₂ := by
+  rw [show r = colimit.ι R (Concrete.indexRepColimit R r) _ by
+    erw [Concrete.ι_repColimit_eq], rep_smul, rep_smul, rep_smul, smulColimit.smul_add]
+
+lemma add_smul (r₁ r₂ : colimit (C := RingCat) R) (m : colimit (C := AddCommGrp) M.presheaf) :
+    colimitSMulColimit (r₁ + r₂) m = colimitSMulColimit r₁ m + colimitSMulColimit r₂ m := by
+  classical
+  let O : Finset Cᵒᵖ :=
+    {  Concrete.indexRepColimit R r₁, Concrete.indexRepColimit R r₂ }
+  let H : Finset ((X : Cᵒᵖ) ×' (Y : Cᵒᵖ) ×' (_ : X ∈ O) ×' (_ : Y ∈ O) ×' (X ⟶ Y)) := {}
+  let j : Cᵒᵖ := IsFiltered.sup O H
+  have eq₁ : r₁ = colimit.ι R j
+      (R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₁)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₂ : r₂ = colimit.ι R j
+      (R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₂)) := by
+    erw [colimit.w_apply, Concrete.ι_repColimit_eq]
+  have eq₃ : r₁ + r₂ = colimit.ι R j
+      (R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₁) +
+       R.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit R r₂)) := by
+    rw [map_add]
+    erw [colimit.w_apply, colimit.w_apply, Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+  rw [eq₃]
+  conv_rhs => rw [eq₁]; rhs; rw [eq₂]
+  rw [rep_smul, rep_smul, rep_smul, smulColimit.add_smul]
+
+lemma zero_smul (m : colimit (C := AddCommGrp) M.presheaf) :
+    colimitSMulColimit (0 : colimit (C := RingCat) R) m = 0 := by
+  let c : Cᵒᵖ := (inferInstance : IsFiltered Cᵒᵖ).2.some
+  rw [show (0 : colimit (C := RingCat) R) = colimit.ι R c 0 by rw [map_zero], rep_smul,
+    smulColimit.zero_smul]
+
+end colimitSMulColimit
+
+noncomputable instance : Module (colimit (C := RingCat) R) (colimit (C := AddCommGrp) M.presheaf) where
```

Comment:
@alreadydone Can you take a look at this file to see if this is what you meant by "some API's about colimit of modules over colimit of rings". Thanks!